internal/bminventory/inventory.go

```diff
@@ -491,10 +491,22 @@ func (b *bareMetalInventory) RegisterClusterInternal(
 	url := installer.V2GetClusterURL{ClusterID: id}
 
 	log := logutil.FromContext(ctx, b.log).WithField(ctxparams.ClusterId, id)
-	log.Infof("Register cluster: %s with id %s and params %+v", swag.StringValue(params.NewClusterParams.Name), id, params.NewClusterParams)
+
+	// Masked the Pull Secret to prevent its value from being displayed in the logs.
+	clusterParamsNoPullSecret := *params.NewClusterParams
+	if clusterParamsNoPullSecret.PullSecret != nil {
+		*clusterParamsNoPullSecret.PullSecret = "***"
+	}
+
+	jsonNewClusterParams, err := json.Marshal(clusterParamsNoPullSecret)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Infof("Register cluster: %s with id %s and params %s", swag.StringValue(params.NewClusterParams.Name), id, jsonNewClusterParams)
 	success := false
 	cluster := &common.Cluster{}
-	var err error
```

Comment:
why ?

Reply:
@danmanor bc line 501

Reply:
can you send it here please ?

Reply:
this line : `jsonNewClusterParams, err := json.Marshal(clusterParamsNoPullSecret)`
But I'm going to move my code to external method according to Elior's suggesion